Makefile

```diff
@@ -3,6 +3,8 @@ SHELL := /usr/bin/env bash
 GO_TEST_PACKAGES :=./pkg/... ./cmd/...
 GO_BUILD_BINDIR := bin
 
+export GOTOOLCHAIN=go1.23.4
+
```

Comment:
Can we remove this from the PR? I feel like figuring out what we want to do with GOTOOLCHAIN is deserving of its own PR and discussion.

Reply:
This was the first thing that this PR needed because any vendor update (ie any modification to the go.mod file) was being accompanied by an automatic go version update 😄 

I'm trying to think if there's a good follow up to capture that'll make us come back to this? Maybe an issue that says "let's revisit this"?

Reply:
That's because of your environment, not anything intrisically related to bumping openshift/api and kubernetes repos.

Reply:
So even for my environment, when I was downloading and installing go1.23.4 binary explicitly, when I checked go version it was automatically downloading and installing go1.23.5 by replacing go1.23.4 (which would then update the gotoolchain in the go.mod file). We talked about it in this [slack thread](https://redhat-internal.slack.com/archives/C080C4VDMDZ/p1740064903531859?thread_ts=1739996798.737199&cid=C080C4VDMDZ) too and finally came to this conclusion. 

However, looks like I've managed to pin the go version to 1.23.4 locally for myself. Let me remove this change and see if I can avoid the unexpected go version patch update.

Reply:
I just looked again and you're right, we most likely don't need this change in the Makefile (go.mod file already has the go version pinned), removing it..